code-challenges.js

```diff
@@ -41,22 +58,60 @@ const padres1998WorldSeriesRuns = [6, 3, 5, 3]
 // Expected output: 9
 
 // Pseudo code:
+// INPUT: one function addUp and 2 variables aka. arguments will be passed on to my code and it shall bring the solution to us all.
+// OUTPUT: as expected should be the length of two arrays summed up and joind like it was done in one movie that-shall-not-be-named (becuase it is just to desturbing), and than sang in the song by Tom Cardy "Red Flags". 
+// PROCESS: "This function is not self invoking, it will require an input." -me
+
+const addUp = (str1, str2) => {
+    return str1.length + str2.length
+}
+console.log(addUp(padres1984WorldSeriesRuns, padres1998WorldSeriesRuns))
 
 // --------------------3) Create the code that will reverse the letters of a string using the test variable provided below.
 
 const currentCohort = "Foxtrot 2023"
 // Expected output: "3202 tortxoF"
 
 // Pseudo code:
+//INPUT: i was stressing a lot but what i realized is that the less i stress the more it makes sense. As it turns out the function just takes in the variable and makes it go to work, and console.log makes me go to work. So the input is THE VARIABLE
+// OUTPUT: it is the reverse string
+// PROCESS: the first one will be just a console.log() and the second one is wil be: "This function is not self invoking, it will require an input." -me
+// NOTES && QUASTIONS:
+// THIS didn't work:
+// console.log(currentCohort.split(""))
+// console.log(currentCohort.reverse(""))
+// console.log(currentCohort.join(""))
+// Why NOT???
+// AND
+// ??? since i turned the string into an array and than reversed and joind it, why did it come back as a string not stayed as an array???
```

Comment:
Some methods are accessors while others a mutators. Accessors will not change the original data, while mutators will make the change permanent. The `.split()` method is an accessor, so even if you use it on `currentCohort`, it doesn't change the original string. So when you get to line 82, it is trying to use `.reverse()` on a string, which will not work.

It comes back as a string because you use `.split` to turn the string into an array, then `.reverse` to reverse the array, then `.join` to turn the array back into a string.